packages/application-shell/src/components/application-page-title/application-page-title.tsx

```diff
@@ -0,0 +1,127 @@
+import type { TEnhancedLocation } from '@commercetools-frontend/browser-history';
+
+import { useLayoutEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import startCase from 'lodash/startCase';
+import capitalize from 'lodash/capitalize';
+
+type Breadcrumb = {
+  suffix: string;
+  paths: string[];
+};
+type Props = {
+  /**
+   * Allow to render a custom page title.
+   * Overrides default page title with the format:
+   * <?resource_name(manually inputed)> - <?page_location> - <application_identifier> - <project_key> - Merchant Center
+   */
+  renderPageTitle?: string[];
+};
+
+const maxTitleCharLength = 24;
+
+const compilePath = (paths: string[], hasTruncatedPaths = false): string[] => {
+  if (hasTruncatedPaths) {
+    return [...paths.slice(0, 1), '...', ...paths.slice(1)];
+  }
+  return paths;
+};
+
+const getLimitedPaths = (
+  paths: string[],
+  hasTruncatedPaths = false
+): string[] => {
+  // We want to keep a minimum of 2 paths
+  if (paths.length <= 2) {
+    return compilePath(paths, hasTruncatedPaths);
+  }
+
+  // Calculate the length projection
+  const pathsProjection = paths.join('-');
+
+  // If the length exceeds the max allowed length, we need to remove one of the paths
+  // from the list, except from the first and last.
+  if (pathsProjection.length > maxTitleCharLength) {
+    return getLimitedPaths(
+      [
+        // we always remove the second item
+        ...paths.slice(0, 1),
+        ...paths.slice(2),
+      ],
+      true
+    );
+  }
+  return compilePath(paths, hasTruncatedPaths);
+};
+
+const isStaticPath = (path: string) => {
+  const staticPaths = ['account', 'login'];
+  return staticPaths.includes(path);
+};
+
+/**
+ * Allows to extend the page title.
+ * When provided, the array items will be used as a prefix for the default title this component will generate.
+ *
+ * It will follow a pattern like this:
+ * <renderPageTitle.join(' - ')> - <?page_location> - <application_identifier> - <project_key> - Merchant Center
+ *
+ */
+
+const compilePageTitle = <Query extends {}>(
+  location: TEnhancedLocation<Query>
+): Breadcrumb => {
+  const [, projectKeyOrStaticPath, ...locationPaths] =
```

Comment:
Since we only look for the project key and the entry point, we only need to pick the first two:

```suggestion
  const [, projectKeyOrStaticPath, entryPointUriPath] =
```

<img width="254" alt="image" src="https://user-images.githubusercontent.com/1110551/186464254-cf38138a-3d9c-400e-9ad8-5d7e064626f5.png">